pennylane/devices/qubit/measure.py

```diff
@@ -86,17 +88,21 @@ def csr_dot_products(
         TensorLike: the result of the measurement
     """
     total_wires = len(state.shape) - is_state_batched
-    Hmat = measurementprocess.obs.sparse_matrix(wire_order=list(range(total_wires)))
 
     if is_state_batched:
```

Comment:
I thought of that too. It should be trivial and beneficial to loop over a batched state's columns, but it's also easy to generalize the code to handle broadcasting along one dimension. I had issues creating a test case that hits that part of the code. For instance, returning `[qml.expval(H), qml.expval(H)]` does not work. If you know an example on top of your head, providing it would be helpful.

Reply:
to create a batched state, you just need a batched op. something like:
```python
def circuit(x):
    qml.RX(x, 0)
    qml.CNOT([0, 1])
    return qml.expval(qml.Hamiltonian([1.1, 2.2], [qml.PauliZ(0), qml.PauliX(0)]))

circuit(np.array([0.5, 0.8]))
```
when we hit the batched RX, the state will become batched automatically.

Reply:
Thanks. I was trying to make this work with `qml.UCCSD`, but didn't work. Can qnodes batch over 2d-arrays? Maybe not?

Reply:
templates and parameter broadcasting coverage is hit and miss.  We can look into adding support to a particular template if you want it.  Basically, we would have to set the `ndim_params` property on the template, and potentially some other fixes.